esp/esp/program/models/__init__.py

```diff
@@ -1319,6 +1327,31 @@ def getLastProfile(user):
     getLastProfile.depend_on_row('program.RegistrationProfile', lambda profile: {'user': profile.user})
     getLastProfile = staticmethod(getLastProfile) # a bit annoying, but meh
 
+    @cache_function
+    def get_last_program_with_profile(user):
+        """
+        Return the program for which the user most recently wrote a profile.
+
+        Look up the profile most recently written by the user, skipping
+        profiles not associated with a program, and return the associated
+        program. If no such programs exist, return None. Used as an
+        approximation of the most recent program attended by the user, which is
+        also often a currently running program.
+        """
+        try:
+            return (
+                RegistrationProfile.objects
+                .filter(user__exact=user, program__isnull=False)
+                .select_related('program')
+                .latest('last_ts')
+                .program
+            )
+        except:
```

Comment:
this should catch a more specifc error (likely `RegistrationProfile.DoesNotExist`)